packages/combobox/src/ComboboxOption/ComboboxOption.tsx

```diff
@@ -74,19 +140,28 @@ export const InternalComboboxOption = React.forwardRef<
         aria-label={displayName}
         darkMode={darkMode}
         className={cx(
-          comboboxOptionBaseStyle,
           comboboxOptionSizeStyle(size),
+          {
+            [multiselectIconPosition]: multiselect && !withIcons,
+            [multiselectIconLargePosition]:
+              multiselect && !withIcons && size === ComboboxSize.Large,
+          },
           className,
         )}
         onClick={handleOptionClick}
         onKeyDown={handleOptionClick}
       >
-        <OptionContent
-          disabled={disabled}
-          displayName={displayName}
-          glyph={glyph}
-          isSelected={isSelected}
-        />
+        <InputOptionContent
+          leftGlyph={multiselect ? multiSelectLeftGlyph : singleSelectLeftGlyph}
+          rightGlyph={
+            multiselect ? multiSelectRightGlyph : singleSelectRightGlyph
+          }
```

Comment:
nit: I feel like these ternary checks should happen up on ~L131 and assign to only 2 vars `leftGlyph` and `rightGlyph`